Update Intstab Manual (#9)
TU-Darmstadt-APQ/RedPitaya-IntStab: manual/manual.tex

\section{Setting up the software}

\index{PyRPL}\marginlabel{Install PyRPL} 

> The RedPitaya is operated with the Python software package PyRPL. You need a proper Python installation in order to run the software. For Windows OS, use \href{https://www.anaconda.com/}{Anaconda Python}. Create a virtual environment for PyRPL. As the PyRPL branch used is not longer maintained, the dependencies clash with the latest version of Python and the environment setup process is outdated. Therefore, to begin, an auxiliary environment is created. Starting with Python version 3.11 and Anaconda Navigator 2.4.0 is recommended. To create the correct environment, run

Reviewer: ```suggestion
The RedPitaya is operated with the Python software package \href{https://github.com/pyrpl-fpga/pyrpl}{PyRPL}. You need a proper Python installation in order to run the software. For Windows OS, use \href{https://www.anaconda.com/}{Anaconda Python}. Create a virtual environment for PyRPL. As the \href{https://github.com/pyrpl-fpga/pyrpl/tree/external_pid_pause}{external\_pid\_pause PyRPL branch} used is not longer maintained, the dependencies clash with the latest version of Python and the environment setup process is outdated. Therefore, to begin, an auxiliary environment is created. Starting with Python version 3.11 or higher installed on the system and Anaconda Navigator 2.4.0 is recommended. To create a compatible Python 3.9 auxiliary environment, run
```

\begin{tcolorbox}
	\begin{verbatim}
		conda create -n aux-env python=3.9 numpy=1.19
		  pyqtgraph=0.12 pandas=1.3 scipy paramiko nose pip 
		  pyqt qtpy=1.9.0 pyyaml ipywidgets notebook
		conda activate pyrpl
		pip uninstall pyopenssl cryptography
		pip install pyopenssl==22.0.0 cryptography==36.0.2
		pip install quamash
		pip install scp
	\end{verbatim}
\end{tcolorbox}

Note that the support for Python 3.9 ends in 2025. Activate the environment via

\begin{tcolorbox}
	\begin{verbatim}
		conda activate aux-env
	\end{verbatim}
\end{tcolorbox}

> Get \textit{pyrpl-env.yaml} from \href{https://github.com/TU-Darmstadt-APQ/RedPitaya-IntStab/tree/master/pyrpl}{https://github.com/TU-Darmstadt-APQ/RedPitaya-IntStab/tree/ master/pyrpl} and run (in the directory of the file) 

Reviewer: ```suggestion
Get \textit{pyrpl-env.yaml} from \href{https://github.com/TU-Darmstadt-APQ/RedPitaya-IntStab/tree/master/pyrpl}{https://github.com/TU-Darmstadt-APQ/RedPitaya-IntStab/tree/ master/pyrpl} and run (in the directory of the file).
```

	\end{verbatim}
\end{tcolorbox}

It might be necessary to install certain Microsoft Visual Studio C++ tools in order to install the packages. \\

> \textbf{Note:} The branch 'external\_pid\_pause' is not the latest stable version of PyRPL. It will not be updated. If the sample and hold feature (see Section ~\ref{sec:pyrpl_settings})  is not used, consider installing the 'master' branch in a new environment. The environment created with the instructions above only works for the  'external\_pid\_pause' branch. 

Reviewer: ```suggestion
\textbf{Note:} The branch 'external\_pid\_pause' is not the latest stable version of PyRPL. It will not be updated. If the sample and hold feature (see Section ~\ref{sec:pyrpl_settings}) is not used, consider installing the 'master' branch in a new environment. The environment created with the instructions above only works for the  'external\_pid\_pause' branch. 
```

Reviewer: Removed a redundant whitespace character.

\subsection{PyRPL settings} \label{sec:pyrpl_settings}

\index{Input filters}\marginlabel{Input filters}
PyRPL allows to set four digital filters as input filters for the PI. Use this feature if the bandwidth of the photodetector is much larger than the control bandwidth.

\index{Sample and hold}\marginlabel{Sample and hold}

> The achievable transient times are limited by the bandwidth of the control loop. For applications that require control of the light fields on shorter time scales, such as short light pulses,the sample and hold featurecan be used. In order to do so, set 'pause\_gains' to 'pi' in the 'pid' module. The sample and hold feature can be activated manually by ticking the 'paused' checkbox. Note, that the features are only available if the branch 'external\_pid\_pause' is installed.

Reviewer: ```suggestion
The achievable transient times are limited by the bandwidth of the control loop. For applications that require control of the light fields on shorter time scales, such as short light pulses, the sample and hold feature can be used. In order to do so, set 'pause\_gains' to 'pi' in the 'pid' module. The sample and hold feature can be activated manually by ticking the 'paused' checkbox. Note, that the features are only available if the branch 'external\_pid\_pause' is installed.
```
